apps/hash-graph/lib/graph/src/store/validation.rs

```diff
@@ -52,33 +137,61 @@ where
                 .change_context(QueryError)?;
         }
 
-        self.store
+        Ok(())
+    }
+}
+
+impl<S, A> OntologyTypeProvider<DataType> for StoreProvider<'_, S, A>
+where
+    S: Read<DataTypeWithMetadata, Record = DataTypeWithMetadata>,
+    A: AuthorizationApi + Sync,
+{
+    #[expect(refining_impl_trait)]
+    async fn provide_type(
+        &self,
+        type_id: &VersionedUrl,
+    ) -> Result<Arc<DataType>, Report<QueryError>> {
+        let data_type_id = DataTypeId::from_url(type_id);
+
+        if let Some(cached) = self.cache.data_types.get(&data_type_id).await {
+            return cached;
+        }
+
+        if let Err(error) = self.authorize_data_type(data_type_id).await {
+            self.cache.data_types.deny(data_type_id).await;
+            return Err(error);
+        }
+
+        let type_ = self
+            .store
             .read_one(
                 &Filter::<S::Record>::for_versioned_url(type_id),
                 Some(&QueryTemporalAxesUnresolved::default().resolve()),
```

Comment:
Noticed an error here: this must not be `default`. Could you drive-by this to the following snippet:
```rust
QueryTemporalAxesUnresolved::DecisionTime {
    pinned: PinnedTemporalAxisUnresolved::new(None),
    variable: VariableTemporalAxisUnresolved::new(None, None),
}
```

Also applies to the other two implementations

Reply:
fixed in [204910b](https://github.com/hashintel/hash/pull/3633/commits/204910b84646c5601c7c81c274faba86df980b30)

Reply:
Great, thank you!